sw/device/lib/dif/dif_csrng.h

```diff
@@ -0,0 +1,550 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_CSRNG_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_CSRNG_H_
+
+/**
+ * @file
+ * @brief <a href="/hw/ip/csrng/doc/">CSRNG</a> Device Interface Functions
+ */
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * This API implements an interface for the CSRNG hardware.
+ *
+ * The API follows the naming conventions used in NIST SP 800-90Ar1:
+ *
+ * - Instantiate.
+ * - Update
+ * - Reseed
+ * - Generate
+ * - Uninstantiate
+ *
+ * The seed used for instantiation can either be provided directly in
+ * hardware by an entropy source module; or directly by software. See
+ * `dif_csrng_instantiate()` and `dif_csrng_instantiate_source` for
+ * more details.
+ *
+ * The following sequence of operations are required to initialize
+ * and generate criptographic entropy from the CSRNG module:
+ *
+ * - `dif_csrng_init()`
+ * - `dif_csrng_configure()`
+ * - `dif_csrng_instantiate()`
+ * - `dif_csrng_generate()`
+ * - `dif_csrng_unistnantiate()`
+ *
+ * The following functions can be used for reseed and update operations:
+ *
+ * - `dif_csrng_reseed()`
+ * - `dif_csrng_update()`
+ *
+ * The following utility functions are available to poll the state of
+ * the hardware:
+ *
+ * - `dif_csrng_get_cmd_interface_status()`
+ * - `dif_csrng_get_output_status()`
+ *
+ * Please see the following documentation for more information:
+ * https://docs.opentitan.org/hw/ip/csrng/doc/
+ *
+ * Remaining work:
+ *
+ * - Add error status interface.
+ * - Add internal state control and debug interface.
+ */
+
+/**
+ * A toggle state: enabled, or disabled.
+ *
+ * This enum may be used instead of a `bool` when describing an enabled/disabled
+ * state.
+ */
+typedef enum dif_csrng_toggle {
+  /*
+   * The "enabled" state.
+   */
+  kDifCsrngToggleEnabled,
+  /**
+   * The "disabled" state.
+   */
+  kDifCsrngToggleDisabled,
+} dif_csrng_toggle_t;
```

Comment:
Not sure this matters: the ordering of these enum values assings Enabled = 0 and Disabled = 1.

Reply:
This was part of the boilerplate code generated by `util/make_new_dif.py`. I have no objections to updating the ordering, we should update the template as well.